src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -40,14 +40,27 @@ import LogOutPreviousUserPage from '../../../pages/LogOutPreviousUserPage';
 import networkPropTypes from '../../../components/networkPropTypes';
 import {withNetwork} from '../../../components/OnyxProvider';
 
+let currentUserEmail;
 Onyx.connect({
-    key: ONYXKEYS.MY_PERSONAL_DETAILS,
+    key: ONYXKEYS.SESSION,
     callback: (val) => {
+        // When signed out, val is undefined
         if (!val) {
             return;
         }
 
-        const timezone = lodashGet(val, 'timezone', {});
+        currentUserEmail = val.email;
+    },
+});
+
+Onyx.connect({
+    key: ONYXKEYS.PERSONAL_DETAILS,
+    callback: (val) => {
+        if (!val) {
+            return;
+        }
+
+        const timezone = lodashGet(val, currentUserEmail, 'timezone', {});
```

Comment:
Looks like this needed to be `lodashGet(val, [currentUserEmail, 'timezone'], {});`